### PR TITLE
Javadoc typos and clarifications

### PR DESCRIPTION
**Pull request by viliam-durina**
All changes are javadoc-only

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/core/IMap.java [hunk 1]
```diff
@@ -2109,60 +2109,60 @@ boolean tryLock(K key, long time, TimeUnit timeunit, long leaseTime, TimeUnit le
...
      * </ol>
-     * If the EntryProcessor implements the Offloadable and ReadOnly interfaces the processing will be offloaded to the
+     * If the EntryProcessor implements the Offloadable and ReadOnly interfaces, the processing will be offloaded to the
      * given ExecutorService allowing unblocking the partition-thread. Since the EntryProcessor is not supposed to do
-     * any changes to the Entry the key will NOT be locked for the time-span of the processing. In this case the threading
+     * any changes to the Entry, the key will NOT be locked for the time-span of the processing. In this case the threading
      * looks as follows:
      * <ol>
-     * <li>partition-thread (fetch & lock)</li>
+     * <li>partition-thread (fetch)</li>
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/core/IMap.java]: WHat happened to the &lock?

File: hazelcast/src/main/java/com/hazelcast/core/IMap.java [hunk 2]
```diff
@@ -2187,27 +2187,46 @@ boolean tryLock(K key, long time, TimeUnit timeunit, long leaseTime, TimeUnit le
...
      * capacity.
      *
+     * <h4>Performance note</h4>
+     * <p>Keep the state of {@code entryProcessor}
+     * small, it will be serialized and one copy will be sent to each member.
+     * Additionally, the {@linkplain EntryProcessor#getBackupProcessor() backup
+     * processor} will also be serialized once for each affected partition and
+     * sent to to each backup. For example, this usage is very inefficient; it
+     * will duplicate the entire {@code additions} map once for each member and
+     * once for each partition and backup:
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/core/IMap.java]: You are leaving me on a cliff hanger. You tell something bad is going to happen, but you don't provide a solution! :)

File: hazelcast/src/main/java/com/hazelcast/core/IMap.java (hunk 1 above)
Comment (viliam-durina) [reply hazelcast/src/main/java/com/hazelcast/core/IMap.java]: I think it's not supposed to be there. On line 2123 it says "the key will NOT be locked" (emphasis was not added), also there's no unlock. I didn't check the actual implementation though.

Comment (viliam-durina) [reply hazelcast/src/main/java/com/hazelcast/core/IMap.java]: Direct message from Ahmet Mırçık Today at 11:36
>Hi, your change is correct, there is no locking for read-only entry processors

**Comment by viliam-durina** [comment]
I merge with one approval since the change is javadoc-only and non-trivial details were confirmed.

**Pull request merged**